scripts/market_parameters.ipynb

```diff
@@ -179,89 +282,321 @@
+    "    (L0, L1, earnings) = fpmm_trade_all(trades, L0, L1, 0.02, 0)\n",
+    "    p = L0/(L0+L1)\n",
+    "    expected_value = 0.6814355029609638\n",
+    "    tolerance = 1e-6\n",
+    "    assert abs(expected_value - p) < tolerance, \"Market simulation does not match expected result.\"\n",
+    "    \n",
+    "    # Test simulations\n",
+    "    assert abs(simulate(4, 0, 1.0, 10, 10, 0.02, 0, 1, 0, 1)[1] - 0.6840229885057476) < tolerance, \"Market simulation does not match expected result: 4 win trades\"\n",
+    "    assert simulate(0, 4, 1.0, 10, 10, 0.02, 2, 1, 0, 1)[1] == 0, \"Market simulation does not match expected result: ROI for mech costs > available amount != 0. \"\n",
+    "    assert simulate(0, 4, 1.0, 10, 10, 0.02, 0, 1, 0, 1)[1] == -1, \"Market simulation does not match expected result: ROI for all trades lose != -1.\"\n",
+    "\n",
+    "\n",
+    "def tests2() -> None:\n",
+    "    L0 = 11000000000000000000\n",
+    "    L1 = 9000000000000000000\n",
+    "    trades = [\n",
+    "        (1199999999999994880,1),\n",
+    "        (210820759921632064,1),\n",
+    "        (469987407320034624,1),\n",
+    "        (731356169902985344,1)\n",
+    "    ]\n",
+    "    (L0, L1, earnings) = fpmm_trade_all(trades, L0, L1, 0.02, 0)\n",
+    "    print(L0)\n",
+    "    print(L1)\n",
+    "    print(\"\")\n",
+    "\n",
+    "    print(fpmm_trade(amount=1199999999999994880, option=1, L0=11000000000000000000, L1=9000000000000000000, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
+    "    print(\"9728773584905664512, 10175999999999994880\")\n",
+    "    print(\"Ok\")\n",
+    "\n",
+    "    print(fpmm_trade(amount=210820759921632064,  option=1, L0=9728773584905664512,  L1=10175999999999994880, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
+    "    print(\"9935377929628863936, 9964391963869474816\")\n",
+    "    print(\"Fails\")\n",
+    "\n",
+    "    print(fpmm_trade(amount=469987407320034624,  option=1, L0=9935377929628863936,  L1=9964391963869474816, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
+    "    print(\"9496421439632642048, 10424979623043108736\")\n",
+    "    print(\"Ok\")\n",
+    "\n",
+    "    print(fpmm_trade(amount=731356169902985344,  option=1, L0=9496421439632642048,  L1=10424979623043108736, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
+    "    print(\"8885531199588972544, 11141708669548034304\")\n",
+    "    print(\"Ok\")\n",
+    "\n",
+    "\n",
+    "    print(\"Second test\")\n",
+    "    print(fpmm_trade(amount=387463502366054272,  option=1, L0=11000000000000000000,  L1=9000000000000000000, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
+    "    print(\"10554692557571284992, 9379714232318733184\")\n",
+    "    print(\"Ok\")\n",
+    "\n",
+    "    print(fpmm_trade(amount=509961030758124800,  option=1, L0=10554692557571284992,  L1=9379714232318733184, market_fee=0.02, mech_cost=0, mech_calls_per_trade=0))\n",
```

Comment:
This is wrong. This trade has the mech output with p_yes > p_no so the voted option in this trade is option=0